doc/source/whatsnew/v0.24.0.txt

```diff
@@ -626,7 +626,7 @@ MultiIndex
 
 - Removed compatibility for :class:`MultiIndex` pickles prior to version 0.8.0; compatibility with :class:`MultiIndex` pickles from version 0.13 forward is maintained (:issue:`21654`)
 - :meth:`MultiIndex.get_loc_level` (and as a consequence, ``.loc`` on a :class:``MultiIndex``ed object) will now raise a ``KeyError``, rather than returning an empty ``slice``, if asked a label which is present in the ``levels`` but is unused (:issue:`22221`)
--
+- Fix `TypeError` in Python 3 when creating :class: `MultiIndex` in which some columns are of object type, e.g. when labels are tuples  (:issue:`15457`)
```

Comment:
@dickreuter OK, I think I see your point. What I'm saying is "not all ``object``" dtype give problems, what you are saying is "other ``object`` dtypes than tuples give problems". I think a more precise description then is "in which some levels have mixed types", as I guess happens in your case (no error comes if _all_ labels are ctype items, right?).

Then i think we're ready.

Reply:
I believe in the example I first hit the error there where 3 columns. All of them were dtype and all the rows were homogenous of the same type as well.

Reply:
@toobaz When you say mixed type, is the column name also relevant for that for each for? If yes, then I assume it’s a mixed type because the column names were strings in my case.

Reply:
Your new version is almost ready (see my last comment), but for the records:

> I believe in the example I first hit the error there where 3 columns. All of them were dtype and all the rows were homogenous of the same type as well.

The error only comes out when two elements (in a same column) cannot be ordered, which in Python 3 happens when they are of non-comparable types. If you got an error with columns which homogeneous types, I doubt it was the same error.
Notice that I'm referring to (Python) _types_, not _dtypes_. If you put mixed _types_ in a single column, then its dtype will be just ``object``, but it is still a mixed type column.

> When you say mixed type, is the column name also relevant for that for each for?

No, column names are stored separately and should be unrelated to this bug and to any issue with mixed type columns/levels.